pkg/cloudprovider/azure.go

```diff
@@ -586,17 +589,47 @@ func (a *Azure) getAuthorizer(env azureapi.Environment, cfg *azureCredentialsCon
 		err  error
 	)
 
-	// MSI Override for ARO HCP
-	msi := os.Getenv("AZURE_MSI_AUTHENTICATION")
-	if msi == "true" {
-		options := azidentity.ManagedIdentityCredentialOptions{
+	// Managed Identity Override for ARO HCP
+	managedIdentityClientID := os.Getenv("ARO_HCP_MI_CLIENT_ID")
+	if managedIdentityClientID != "" {
+		klog.Info("Using client certification Azure authentication for ARO HCP")
+		options := &azidentity.ClientCertificateCredentialOptions{
 			ClientOptions: azcore.ClientOptions{
 				Cloud: cloudConfig,
 			},
+			SendCertificateChain: true,
```

Comment:
Out of curiosity, could you comment on why is this required?

Reply:
This is related to 1st party Azure technology but with this, the certificate authentication doesn't require pinning certificates to applications, basically saying "Trust this certificate subject issued from this CA". cc @bennerv

Reply:
Note that it has no averse affect if you don't use SNI authentication with the 1p Azure technology.  The azure-sdk-for-go comment has a [bit more detail:](https://github.com/Azure/azure-sdk-for-go/blob/9591fab0ac0e5e3e2707d6db579331b136fb00ea/sdk/azidentity/client_certificate_credential.go#L45-L48) 
> 	// SendCertificateChain controls whether the credential sends the public certificate chain in the x5c
>	// header of each token request's JWT. This is required for Subject Name/Issuer (SNI) authentication.
>	// Defaults to False.